client/my-sites/checkout/get-thank-you-page-url/index.ts

```diff
@@ -283,6 +284,8 @@ export default function getThankYouPageUrl( {
 			{
 				receiptId: receiptIdOrPlaceholder,
 				siteId: jetpackTemporarySiteId && parseInt( jetpackTemporarySiteId ),
+				fromSiteSlug,
+				redirect_to: redirectTo,
```

Comment:
I think I understand what you are talking about, but please correct me if I'm mistaken. Other than the flows you mentioned above, if a site is not fully connected i.e. `connectAfterCheckout` is false, it would end up here as well. Does it break anything if `fromSiteSlug` ends up empty because they originate from jetpack.com for example?

With this [PR](https://github.com/Automattic/jetpack/pull/39573), it isn't a problem anymore, when we can always pass `connectAfterCheckout` as `true` to make it work. However, for older versions of Jetpack, it would remain an issue until upgraded.

That said, I think we can get rid of `fromSiteSlug` if you think it's a better idea, as it seems only affects whether the license purchased would be automatically assigned or not.

Related code:
https://github.com/Automattic/wp-calypso/blob/472d2abe5ec784bd225a4905235a058f7d1705a4/client/my-sites/checkout/get-thank-you-page-url/index.ts#L244

Reply:
> Other than the flows you mentioned above, if a site is not fully connected i.e. connectAfterCheckout is false, it would end up here as well.

Umm, hmm, depends what you mean..  yes, if we're going to link a user to `/checkout/jetpack/:product_slug`, and we also include 3 query args with it (`connectAfterCheckout` && `adminUrl` && `fromSiteSlug`) then checkout will attempt to connect the site after purchase is complete and then auto activate the product license key (connect-after-checkout flow).  However if we don't send All 3 query args (all with truthy values) then yes, they would end up in regular [siteless checkout](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/get-thank-you-page-url/index.ts#L278) flow (AKA: licensed-based checkout) where checkout will not attempt to connect the site, rather it will just ask the user to activate the product on the site of their choosing. In this scenerio, when going into checkout (since we don't know the user's site, we can't send a valid `fromSiteSlug` value, and also I don't believe we would want to send a `redirect_to` query arg into checkout either...  because we don't know yet their site in which we would want to redirect to..  Instead we would want to wait until after checkout is complete, and then after they have chosen a site to activate their product license key on, it is then when we will know which site they intend the product for, so we can redirect to it then.

> Does it break anything if fromSiteSlug ends up empty because they originate from jetpack.com for example?

No, it won't break anything, but having it there in the siteless checkout flow is weird I would say, and it's misleading because we need and use the siteless checkout flow specifically for scenarios when we don't know the users site, but `fromSiteSlug` implies it should contain the site that the user is coming from, which if we know that, then we shouldn't be sending the user to the siteless checkout flow then.

> With this https://github.com/Automattic/jetpack/pull/39573, it isn't a problem anymore, when we can always pass connectAfterCheckout as true to make it work. However, for older versions of Jetpack, it would remain an issue until upgraded.

Yes true, for older versions of Jetpack, attempting to go through the connect-after-checkout flow when the site is already fully connected would cause a fatal error on the Calypso authorization page, but only when running Calypso in the dev environment (calypso.localhost:3000). The issue only occurred when triggering the Jetpack connection from the Calypso dev environment. However it always worked coming from Calypso production.

Reply:
> Other than the flows you mentioned above, if a site is not fully connected i.e. connectAfterCheckout is false, it would end up here as well.

Yes, if the site is not connected, but we're sending them to checkout with `connectAfterCheckout=false`, then we're sending them the wrong checkout url. Yes, I suppose they would end up in siteless-checkout, but we shouldn't be sending them there to begin with.

Reply:
> That said, I think we can get rid of `fromSiteSlug` if you think it's a better idea, as it seems only affects whether the license purchased would be automatically assigned or not.

Yeah, if we know the user's site, we shouldn't be sending them through the siteless-checkout flow where they have to bother with activating their product license key and so forth. Instead we should be sending them through the regular checkout (`/checkout/:site/:product`). Unless the site is not connected, of course, then we would send them through the connect-after-checkout flow. Then only when we have no idea what their site is at all (i.e.- when landing on https://wordpress.com/stats/purchase when coming from https://jetpack.com/stats), then we should send them through the siteless-checkout flow (of course with no `fromSiteSlug` prop/arg because we don't know the site).
So yeah, in my opinion we should remove `fromSiteSlug` prop from the siteless-checkout flow part.

Reply:
> Instead we should be sending them through the regular checkout (/checkout/:site/:product).

I believe the reason we are using the siteless or site-only flow is for the sake of removing the burdern of user login, which was original introduced by https://github.com/Automattic/wp-calypso/pull/90778. I agree there is no need for user to activate a product when they could be automatically assigned.